l5x/enums.go

```diff
@@ -0,0 +1,322 @@
+package l5x
+
+import (
+	"encoding/xml"
+)
+
+type ExternalAccess int
+
+const (
+	ExternalAccessReadWrite ExternalAccess = iota
+	ExternalAccessReadOnly
+)
+
+var externalAccessNames = []string{"Read/Write", "Read Only"}
+
+func (enum *ExternalAccess) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	return enumUnmarshalXML((*int)(enum), externalAccessNames, d, start)
+}
+func (enum ExternalAccess) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	return enumMarshalXML(int(enum), externalAccessNames, e, start)
+}
+func (enum *ExternalAccess) UnmarshalXMLAttr(attr xml.Attr) error {
+	return enumUnmarshalXMLAttr((*int)(enum), externalAccessNames, attr)
+}
+func (enum ExternalAccess) MarshalXMLAttr(name xml.Name) (xml.Attr, error) {
+	return enumMarshalXMLAttr(int(enum), externalAccessNames, name)
+}
+func (enum ExternalAccess) String() string { return enumToString(int(enum), externalAccessNames) }
+
+type Class int
+
+const (
+	ClassUser Class = iota
+)
+
+var classNames = []string{"User"}
+
+func (enum *Class) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	return enumUnmarshalXML((*int)(enum), classNames, d, start)
+}
+func (enum Class) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	return enumMarshalXML(int(enum), classNames, e, start)
+}
+func (enum *Class) UnmarshalXMLAttr(attr xml.Attr) error {
+	return enumUnmarshalXMLAttr((*int)(enum), classNames, attr)
+}
+func (enum Class) MarshalXMLAttr(name xml.Name) (xml.Attr, error) {
+	return enumMarshalXMLAttr(int(enum), classNames, name)
+}
+func (enum Class) String() string { return enumToString(int(enum), classNames) }
+
+type Radix int
+
+const (
```

Comment:
Do we need to encode the radix, or can that be inferred from the value (ie. `0x` prefixes means it's a hex value, etc)?

Reply:
I don't know. I parse the Radix because it's in the L5X file, but I don't really use it for much.